### PR TITLE
Adding theme property/attribute

### PR DESCRIPTION
**Pull request by manolo**
Needed to fix https://github.com/vaadin/vaadin-dialog/pull/37 via  https://github.com/vaadin/vaadin-dialog/pull/42

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-themable-mixin/21)

### CODE REVIEW

**Comment by tomivirkki** [comment]
*[vaadin-themable-mixin.html, line 17 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themable-mixin/21#-L40OFD4ZkU62mgrbgkA:-L40OFD4ZkU62mgrbgkB:bzhjvuk) ([raw file](https://github.com/vaadin/vaadin-themable-mixin/blob/19147eafecd58719e43e7b205e326d9d37f85d8b/vaadin-themable-mixin.html#L17)):*
> ```HTML
>     static get properties() {
>       return {
>         theme: {
> ```

Public property, needs documentation

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themable-mixin/21)*

**Comment by manolo** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-themable-mixin.html, line 17 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-themable-mixin/21#-L40OFD4ZkU62mgrbgkA:-L40cUAl2LA7K7zip-nb:b-896fix) ([raw file](https://github.com/vaadin/vaadin-themable-mixin/blob/19147eafecd58719e43e7b205e326d9d37f85d8b/vaadin-themable-mixin.html#L17)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary><blockquote>

Public property, needs documentation
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-themable-mixin/21)*

**Comment by jouni** [comment]
What are the plans regarding this? It feels like a good step forward, making the `theme` property/attribute the officially supported way to change the appearance of a “themable component”.

File: vaadin-themable-mixin.html [hunk 1]
```diff
@@ -12,6 +12,18 @@
    */
   Vaadin.ThemableMixin = superClass => class VaadinThemableMixin extends superClass {
 
+    static get properties() {
+      return {
+        /**
+         * built-in variations in Element themes are applied using the theme attribute.
```
Comment (jouni) [inline vaadin-themable-mixin.html]: I’d fix this as “Built-in variations of a component theme are applied using the `theme` attribute.”

**Comment by manolo** [comment]
Yes that was the motivation, not add the theme attribute in all overlay-ed  elements but declare it in this common mixin.

Any kind of solution in overlay-ed elements should not affect the convenience of officially document the `theme` attr here as @jouni mentions.

---

Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-themable-mixin/21#-:-LETVAv2395CG2_MCI0b:b-ciicgv)*

**Comment by platosha** [comment]
Obsolete by #32

**Pull request closed without merging**